resources/services/keystone/services.go

```diff
@@ -0,0 +1,68 @@
+package keystone
+
+import (
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/dihedron/cq-plugin-utils/format"
+	"github.com/dihedron/cq-source-openstack/client"
+
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/services"
+)
+
+func Services() *schema.Table {
+	return &schema.Table{
+		Name:     "openstack_keystone_services",
+		Resolver: fetchServices,
+		Transform: transformers.TransformWithStruct(
+			&services.Service{},
+			transformers.WithSkipFields("Links", "Extra"),
+		),
+	}
+}
+
+func fetchServices(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+
+	api := meta.(*client.Client)
+
+	keystone, err := api.GetServiceClient(client.KeystoneV3)
+	if err != nil {
+		api.Logger().Error().Err(err).Msg("error retrieving client")
+		return err
+	}
+
+	service_types := [5]string{"blockstorage", "compute", "identity", "imageservice", "networking"}
```

Comment:
https://docs.openstack.org/keystone/latest/contributor/service-catalog.html

https://pkg.go.dev/github.com/gophercloud/gophercloud/openstack/identity/v3/catalog

HTTP get all'enpoint pubblico di Keystone relativo al catalogo
unmarshal dentro sytriuutura dati
loop su array di catalog entries.

Reply:
here the catalog API returns an array of [tokens](https://pkg.go.dev/github.com/gophercloud/gophercloud@v1.11.0/openstack/identity/v3/tokens).[CatalogEntry](https://pkg.go.dev/github.com/gophercloud/gophercloud@v1.11.0/openstack/identity/v3/tokens#CatalogEntry) which is part of the identity V2 (deprecated) APIs. 

For the services resource, we just don't tie the call here to a specific `service_type` but we retrieve all services.